### ISSUE TITLE
Please make Build instructions for library dependencies more specific

### ISSUE THREAD

**Issue by LuboGD** [freeorion/freeorion#4940]
I am trying to build FreeOrion 0.5.0.1 from source on AntiX 23 (Debian 12 derivative). All kinds of errors about missing library dependencies, since I had not used said libraries for anything before, as is expected.

I came up with a list of Debian .deb packages which works for me, and should also work for any Debian derived distribution with little or no changes. If any of these packages are not required, please feel free to substitute those with fewer dependencies, as the ones I list below pull in over 100 packages (granted many quite small), and thus over 100 MB of downloads.

sudo apt install build-essential libpython3-dev libboost-filesystem-dev libboost-all-dev libsdl1.2-dev libsdl-image1.2-dev libsdl2-dev libsdl2-ttf-dev libopenal-dev libogg-dev libvorbis-dev libglew-dev

My request is to add this text to the Build Instructions found at:
https://github.com/freeorion/freeorion/blob/master/BUILD.md

Under the heading:

Linux and other Operating Systems:

    Install build and runtime dependencies by the preferred way for the respective OS.

E.g. For Debian and derivatives:
sudo apt install build-essential libpython3-dev libboost-filesystem-dev libboost-all-dev libsdl1.2-dev libsdl-image1.2-dev libsdl2-dev libsdl2-ttf-dev libopenal-dev libogg-dev libvorbis-dev libglew-dev

**Comment by geoffthemedio** [comment]
@o01eg is this reasonable?

**Comment by o01eg** [comment]
Yes, we can add it there.

**Comment by vincele** [comment]
Am I reading this right ?
Mixing SDL 2 and 1.2 ?

on a void linux local build I only see SDL2 linked in.

**Comment by o01eg** [comment]
Yep, SDL1.2 isn't used in the project.

**Comment by vincele** [comment]
I don't have a debian box available to check myself, but doesn't `libboost-all-dev` include `libboost-filesystem-dev` ?

**Comment by vincele** [comment]
Maybe instead of adding a lot into this file, just tell people to have a look at the Dockerfiles in https://github.com/freeorion/freeorion/tree/master/.github/docker

**Comment by o01eg** [comment]
At least we should note that godot is optional dependency.

**Comment by vincele** [comment]
Yes, I stumbled upon references to the godot effort, and wondered if this is still in progress...

**Comment by o01eg** [comment]
@vincele Godot efforts are postponed for a while.
